Library/Homebrew/sbom.rb

```diff
@@ -0,0 +1,354 @@
+        "relationshipType"   => "BUILD_TOOL_OF",
+        "relatedSpdxElement" => "SPDXRef-Package-#{name}-src",
+      },
+    ]
+
+    if compiler_declaration["SPDXRef-Stdlib"].present?
+      base += {
+        "spdxElementId"      => "SPDXRef-Stdlib",
+        "relationshipType"   => "DEPENDENCY_OF",
+        "relatedSpdxElement" => "SPDXRef-Bottle-#{name}",
+      }
+    end
+
+    runtime + patches + base
+  end
+
+  sig { params(runtime_dependency_declaration: T::Array[Hash], compiler_declaration: Hash).returns(T::Array[Hash]) }
+  def generate_packages_json(runtime_dependency_declaration, compiler_declaration)
+    bottle = []
+    if get_bottle_info(source["bottle"])
+      bottle << {
+        "SPDXID"           => "SPDXRef-Bottle-#{name}",
+        "name"             => name.to_s,
+        "versionInfo"      => stable_version.to_s,
+        "filesAnalyzed"    => false,
+        "licenseDeclared"  => "NOASSERTION",
+        "builtDate"        => source_modified_time.to_s,
+        "licenseConcluded" => license,
+        "downloadLocation" => T.must(get_bottle_info(source["bottle"]))["url"],
+        "copyrightText"    => "NOASSERTION",
+        "externalRefs"     => [
+          {
+            "referenceCategory" => "PACKAGE-MANAGER",
+            "referenceLocator"  => "pkg:brew/#{tap}/#{name}@#{stable_version}",
+            "referenceType"     => "purl",
+          },
+        ],
+        "checksums"        => [
+          {
+            "algorithm"     => "SHA256",
+            "checksumValue" => T.must(get_bottle_info(source["bottle"]))["sha256"],
+          },
+        ],
+      }
+    end
+
+    [
+      {
+        "SPDXID"           => "SPDXRef-Archive-#{name}-src",
+        "name"             => name.to_s,
+        "versionInfo"      => stable_version.to_s,
+        "filesAnalyzed"    => false,
+        "licenseDeclared"  => "NOASSERTION",
+        "builtDate"        => source_modified_time.to_s,
+        "licenseConcluded" => license || "NOASSERTION",
+        "downloadLocation" => source["stable"]["url"],
+        "copyrightText"    => "NOASSERTION",
+        "externalRefs"     => [],
+        "checksums"        => [
+          {
+            "algorithm"     => "SHA256",
+            "checksumValue" => source["stable"]["checksum"].to_s,
+          },
+        ],
+      },
+    ] + runtime_dependency_declaration + compiler_declaration.values + bottle
+  end
+
+  sig { returns(Hash) }
+  def to_spdx_sbom
+    runtime_full = []
+
+    if @runtime_dependencies.present?
+      runtime_full = @runtime_dependencies.map do |dependency|
+        bottle_info = get_bottle_info(dependency["bottle"])
+        {
+          "SPDXID"           => "SPDXRef-Package-SPDXRef-#{dependency["name"].tr("/", "-")}-#{dependency["version"]}",
+          "name"             => dependency["name"],
+          "versionInfo"      => dependency["pkg_version"],
+          "filesAnalyzed"    => false,
+          "licenseDeclared"  => "NOASSERTION",
```

Comment:
Might be nice to pull `NOASSERTION` into a symbol and maybe even have a more generic wrapper function for "thing is `present?`, use it, otherwise use `NOASSERTION`".